tracing/src/macros.rs

```diff
@@ -783,6 +783,123 @@ macro_rules! event {
     );
 }
 
+/// Checks whether a span or event is enabled depending on the provided metadata.
+///
+/// This macro is a specialized tool: it is intended to be used prior
+/// to an expensive computation required *just* for that event, but
+/// *cannot* be done as part of an argument to that event, such as
+/// when multiple events are emitted (e.g., iterating over a collection
+/// and emitting an event for each item).
+///
+/// ## Usage
+///
+/// It is possible for `enabled!` to return a false positive or false negative. This might
+/// occur when a subscriber is using a _more specific_ filter than what was (or could be)
+/// provided to `enabled!`. Below are several examples where this might occur:
+///
+/// - If a subscriber is using a filter which may enable a span or event based
+/// on field names, but `enabled!` is invoked without listing field names,
+/// `enabled!` may return a false negative if a specific field name would
+/// cause the subscriber to enable something that would otherwise be disabled.
+/// - If a subscriber is using a filter which enables or disables specific events by
+/// file path and line number,  a particular event may be enabled/disabled
+/// even if an `enabled!` invocation with the same level, target, and fields
+/// indicated otherwise.
+/// - The subscriber can choose to enable _only_ spans or _only_ events, which `enabled`
+/// will not reflect.
+///
+/// ## Examples
+///
+/// `enabled!()` requires a level argument, an optional `target:`
```

Comment:
nit, take it or leave it: i might consider adding a link
```suggestion
/// `enabled!()` requires a [level](crate::Level) argument, an optional `target:`
```